src/geotiff.d.ts

```diff
@@ -0,0 +1,90 @@
+export declare class GeoTIFF {
+  getImage(index?: number): Promise<GeoTIFF.GeoTIFFImage>;
+  parseFileDirectoryAt(offset: number): Promise<GeoTIFF.ImageFileDirectory>;
+  readRasters(options?: GeoTIFF.RasterOptions): Promise<TypedArray>;
+  ifdRequests: { [key: number]: Promise<GeoTIFF.ImageFileDirectory> };
+  dataView: DataView;
+  littleEndian: boolean;
+  cache: any;
+  source: any;
+}
+
+export default GeoTIFF;
+
+/** Typed array of data values, the basic building block of a geotiff */
+export type TypedArray =
+| Uint8Array
+| Int8Array
+| Uint16Array
+| Int16Array
+| Uint32Array
+| Int32Array
+| Float32Array
+| Float64Array;
+
+// A namespace with the same name as the default export is needed to declare additional type exports
+// https://stackoverflow.com/a/51238234/4159809
+export declare namespace GeoTIFF {
+
+  export interface Pool {
+    decode(
+      fileDirectory: FileDirectory,
+      buffer: ArrayBuffer
+    ): Promise<ArrayBuffer>;
+  }
+
+  interface RasterOptions {
+    window?: number[];
+    bbox?: number[];
+    samples?: number[];
+    interleave?: boolean;
+    pool?: Pool;
+    width?: number;
+    height?: number;
+    resampleMethod?: string;
+    enableAlpha?: boolean;
+    signal?: AbortSignal;
+  }
+
+  type RasterData = (TypedArray | TypedArray[]) & {
+    width: number;
+    height: number;
+  };
+
+  interface FileDirectory {
+    ImageDescription: string;
+    SubIFDs?: number[];
+    PhotometricInterpretation?: number;
+  }
+
+  interface ImageFileDirectory {
+    fileDirectory: FileDirectory;
+    geoKeyDirectory: any;
```

Comment:
If there is _some_ shape to these `any` fields, it's better than a blanket `any`. I prefer `Record<string, unknown>` or `Record<string, any>` if appropriate.

Reply:
Thanks, time to take a deep pass